includes/Core/Assets/Assets.php

```diff
@@ -343,15 +343,34 @@ private function get_assets() {
 
 		// Register plugin scripts.
 		$assets = array(
-			new Script(
+			new Script_Data(
 				'googlesitekit-commons',
 				array(
-					'src'          => false,
-					'before_print' => function( $handle ) {
-						$data = wp_scripts()->get_data( $handle, 'data' ) ?: '';
-						$script = 'var googlesitekit = ' . wp_json_encode( $this->get_inline_data() ) . ';';
-
-						wp_scripts()->add_data( $handle, 'data', $script . $data );
+					'global'        => 'googlesitekit',
+					'data_callback' => function () {
+						return $this->get_inline_data();
+					},
+				)
+			),
+			new Script_Data(
+				'googlesitekit-base-data',
+				array(
+					'global'        => '_googlesitekitBase',
```

Comment:
We should probably define a pattern as part of this PR on how to name these globals. Right now the two defined here (`googlesitekit` can be ignored, as it's old) are different as in only one has a `Data` suffix.

Maybe we should set the following rules:
* Every `Script_Data` should have a handle that ends with `-data`.
* Every `Script_Data` should use a global name that matches the handle transformed into camel-case, and prefixed with an underscore (e.g. `googlesitekit-base-data` becomes `_googlesitekitBaseData`).

What do you think?

Reply:
If you agree, let's change the corresponding JS references as well of course.

Reply:
That sounds good, I'll update those 👍

Reply:
Not too many replacements needed.  ✅ https://github.com/google/site-kit-wp/pull/1189/commits/081c145305c843eeb0afd56e5eddb14b2ad60aa7